app/src/main/java/com/github/jorgecastillo/kotlinandroid/functional/predef.kt

```diff
@@ -0,0 +1,73 @@
+package com.github.jorgecastillo.kotlinandroid.functional
+
+import com.github.jorgecastillo.architecturecomponentssample.model.error.*
+import com.github.jorgecastillo.kotlinandroid.di.context.GetHeroesContext
+import kategory.*
+
+typealias Control<F> = MonadControl<F, GetHeroesContext, CharacterError>
```

Comment:
This is an infrastructure file and should generally not live in this project, Simiar as if I were to implement an event bus and use it in the same project.

For the details:

MonadControl is an interface that agreggates the behaviors in MonadReader and MonadError.
The methods contained in there are similar to what you would see in the reader datatypes plus others to deal with errors in the computation context.

We make the AsyncResult implement those but instead of using AsyncResult in our implementation we use the interface MonadControl. This allows us to provide other instances where necessary for example instead of using AsyncResult in tests you can use `BlockingResult` or whatever you want that can implement MonadControl. Note that since there is also a Moinad for Future its async semantics don't leak in our code and it all happens behind flatMap transparent to the user.